### ISSUE TITLE
TypeScript & ES6 & extends

### ISSUE THREAD

**Issue by wgebczyk** [dfahlander/Dexie.js#140]
Based on your example:
```
export class HeroCampDB extends Dexie {
  ...
  constructor() {
    super("HeroCampDB");
    ...
    this.open();
  }
}
```
The problem in on IE Edge 13. The error "Assignment to read-only properties is not allowed in strict mode". It tries to assign [this.name = dbName], but property name on Dexie function is already set.

In general class Zzz extends [function] results in name being used.

Do you plan class ES6 based Dexie code?

**Comment by dfahlander** [comment]
After reading briefly the [ES6 standard](http://www.ecma-international.org/ecma-262/6.0/#sec-class-definitions) for inheritance, I can't see why Edge browser would have that behavior, but I could be missing something.

I tried to reproduce the error on Edge 20 on Windows 10 using the code below but it works for me after turning on es6 support using about:flags in the browser:

```js
"use strict"

function ParentClass(dbName) {
	this.name = dbName;
}

class ChildClass extends ParentClass {
	constructor() {
		super("hejsan");
	}
}

alert ("1");
var x = new ChildClass ();
alert("2");
alert(x.name);
```

**Comment by wgebczyk** [comment]
What's Edge 20?

I'll recheck the issue once again.

**Comment by dfahlander** [comment]
Opening Settings, in the bottom, it sais "Microsoft Edge 20.10240.16384.0"
![image](https://cloud.githubusercontent.com/assets/6791329/10825319/cf101942-7e64-11e5-93e5-7b13017bba7d.png)

**Comment by wgebczyk** [comment]
Oh! Got it. On my Edge 23 it says in 3rd alert "ChildClass".

It might me bug when prototype property leaks to instance.

**Comment by dfahlander** [comment]
See discussion going on https://github.com/dfahlander/Dexie.js/pull/147#issuecomment-153228746

**Comment by wgebczyk** [comment]
Maybe rewrite quickly Dixie to ES6 modules/classes and "Babel" it to ES5?

**Comment by dfahlander** [comment]
I've started the work to use es6 source for Dexie in branch es6.
Initially I will only take advantage of the es6 module system but keep everything else.

A simple npm-based build script (no gulp pr grunt) will use rollup and uglify to take what's in src/Dexie.js and transpile to umd module in dist (pretty much same module format as today, but more 'standardized').

Also I added main:esnext entry in package.json so that es6 systems will pick the es6 main file instead of the transpiled one.

It should be possible to test it out already in the es6 branch. Build using: **npm run build**.
